### ISSUE TITLE
Dependency errors when runnin API in Docker

### ISSUE THREAD

**Issue by cferreirasuazo** [w4af/w4af#630]
When I try to run the api using Docker from main branch, the next error shows up: 

```
w4af's requirements are not met, one or more third-party libraries need to be installed.

On Debian Bookworm systems please install the following operating system packages before running the pip installer:
    sudo apt-get -y install libffi-dev python3-pip build-essential libsqlite3-dev python3-setuptools libxml2-dev python3-dev

A script with these commands has been created for you at /tmp/w4af_dependency_install.sh
```
I added those dependencies in the Dockerfile but the error persist 

the next solution it's working for me: https://github.com/codders/w4af/tree/feat/w4af_api-support

**Comment by codders** [comment]
Hi Cristhian,

What happens if you run the command provided?

Thanks,

Arthur

Cristhian Antonio Ferreira Suazo ***@***.***> schrieb am Mi.,
6. Sept. 2023, 17:24:

**Comment by cferreirasuazo** [comment]
What do you mean with the code provided? You mean the code in
https://github.com/codders/w4af/tree/feat/w4af_api-support. The API works
properly.

**Comment by codders** [comment]
Hi Cristhian,

@security-companion is investigating this. Your issue is related to #338, and the new pull request #634 should resolve the problem.

When #634 is merged and #338 is closed, we can review and see if you still have an issue.

**Comment by cferreirasuazo** [comment]
Alright thank you.

**Comment by codders** [comment]
Hi @cferreirasuazo ,

#338 is now closed (resolved by #644), but the changes I made there don't address exactly your issue. Can you pull the latest code and try again to see if you have the same problem?

You say that you changed the Dockerfile to add the command:

```
sudo apt-get -y install libffi-dev python3-pip build-essential libsqlite3-dev python3-setuptools libxml2-dev python3-dev
```

Can you see in the Docker build output whether the command succeeds? If you run the command and the command succeeds, the system shouldn't ask you to reinstall those libraries. If you're still having the problem, please share the Dockerfile with me in a Gist and I will try and reproduce it here.

**Comment by cferreirasuazo** [comment]
Hi, When I try to run the dockerfile this message shows up,

https://gist.github.com/cferreirasuazo/9b5d572aa6e89ec73df6f9d6a259fff1
It just need to change the python version to 3.11

**Comment by codders** [comment]
Can you make a pull request for your change?

Thanks,

Arthur

Cristhian Antonio Ferreira Suazo ***@***.***> schrieb am Di.,
12. Sept. 2023, 21:51:

**Comment by codders** [comment]
Okay. #661 does some things I'm not comfortable with. Removing the required packages is not quite the right approach here.

The reason you see this error message in the docker image is that the docker image is a composite of a couple of different images. The `python-deps` image installs the python dependencies and copies them over to the `runtime` image (similarly the `node-deps` image for the node dependencies). The `runtime` image doesn't have the dependencies installed, because it doesn't need them once the python packages have all been built and installed.

So, to resolve your issue now and quickly, I am opening #662. This basically does the same thing, but without breaking the dependency-checking mechanism. Hopefully @security-companion can approve this and we'll get it merged.

In the meantime I will close your #661 as #662 replaces it, and this should resolve your issue.

**Comment by codders** [comment]
Should now be fixed - #662 has been merged. Please fetch the latest code and retry.

**Comment by cferreirasuazo** [comment]
Hello @codders. Yes, it's working now. thanks
